pallets/file-system/src/lib.rs

```diff
@@ -364,13 +405,49 @@ pub mod pallet {
     impl<T: Config> Pallet<T> {
         #[pallet::call_index(0)]
         #[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-        pub fn create_bucket(_origin: OriginFor<T>) -> DispatchResult {
-            todo!()
+        pub fn create_bucket(
+            origin: OriginFor<T>,
+            msp: T::AccountId,
```

Comment:
`msp_id` might be confusing as we use that naming for the unique, randomnly generated ID that BSPs and MSPs get when signing up. I think to avoid that confusion in this case `msp_account` might be more suitable

Reply:
After discussion with @ffarall, we would be changing all parameters and storage refering to providers to use the Ids (generated) instead of the base account Ids.

Reply:
Although, I want to do this refactor in another PR to not introduce too much change in a single one.

Reply:
Changing all `T::AccountId` to `ProviderIdFor<T>`